### PR TITLE
partially decompile pokemon_summary_screen

### PR DESCRIPTION
**Pull request by golemgalvanize**
Done up to sub_81C3D54. Decided to switch to a different file

### CODE REVIEW

File: src/pokemon_summary_screen.c [hunk 1]
```diff
@@ -0,0 +1,3680 @@
...
+
+struct contestMove
+{
+    u8 effect;
+    u8 type;
+    u8 comboID;
+    u8 combo[4];
+};
+
+struct contestEffect
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Struct names should be capitalized.

File: src/pokemon_summary_screen.c [hunk 2]
```diff
@@ -0,0 +1,3680 @@
...
+
+struct pokeSummary
+{
+    u16 species; // 0x0
+    u16 species2; // 0x2
+    u8 isEgg; // 0x4
+    u8 level; // 0x5
+    u8 ribbons; // 0x6
+    u8 unk7; // 0x7
+    u8 altability; // 0x8
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Use camelCase.

File: src/pokemon_summary_screen.c [hunk 3]
```diff
@@ -0,0 +1,3680 @@
...
+void sub_81BFAD0(void)
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+void sub_81BFAE4()
+{
+    while (1)
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: while(TRUE)

File: src/pokemon_summary_screen.c [hunk 4]
```diff
@@ -0,0 +1,3680 @@
...
+    InitBgsFromTemplates(0, &gUnknown_0861CBB4, 4);
+    SetBgTilemapBuffer(1, &gUnknown_0203CF1C->unkTilemap2);
+    SetBgTilemapBuffer(2, &gUnknown_0203CF1C->unkTilemap1);
+    SetBgTilemapBuffer(3, &gUnknown_0203CF1C->unkTilemap0);
+    sub_8121E10();
+    schedule_bg_copy_tilemap_to_vram(1);
+    schedule_bg_copy_tilemap_to_vram(2);
+    schedule_bg_copy_tilemap_to_vram(3);
+    SetGpuReg(0, 0x1040);
+    SetGpuReg(0x50, 0);
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Use the definitions from io_reg.h for these.

File: src/pokemon_summary_screen.c [hunk 5]
```diff
@@ -0,0 +1,3680 @@
...
+        return 1;
+    }
+    return 0;
+}
+
+void sub_81C0098(struct Pokemon *poke)
+{
+    if (gUnknown_0203CF1C->unk40BD == 0)
+    {
+        struct Pokemon *pokeMons = gUnknown_0203CF1C->unk0->mons;
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: This should be named something more descriptive, like partyMon if they're in the party.

File: src/pokemon_summary_screen.c [hunk 6]
```diff
@@ -0,0 +1,3680 @@
...
+        *poke = pokeMons[gUnknown_0203CF1C->unk40BE];
+    }
+    else
+    {
+        struct BoxPokemon *boxMons = gUnknown_0203CF1C->unk0->boxMons;
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: mons -> mon

Comment (golemgalvanize) [reply src/pokemon_summary_screen.c]: it's a pointer to the whole section of mons, that's why it's mons and not mon. Unless I'm being stupid and there's another reason why you want it renamed

Comment (Diegoisawesome) [reply src/pokemon_summary_screen.c]: The plural of Pokemon is Pokemon...

Comment (golemgalvanize) [reply src/pokemon_summary_screen.c]: disregard me then I suck cocks

File: src/pokemon_summary_screen.c [hunk 7]
```diff
@@ -0,0 +1,3680 @@
...
+void sub_81C38C0();
+void sub_81C3B08(u8 a);
+void sub_81C3E2C(u8 a);
+
+void SetBgAttribute(u8 bg, u8 attributeId, u8 value);
+
+bool8 sub_81BFB10();
+u8 sub_81B1250();
+
+union unkUnion
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Unions should have the same capitalization as structs (PascalCase).

File: src/pokemon_summary_screen.c [hunk 8]
```diff
@@ -0,0 +1,3680 @@
...
+    u8 unk3E[9]; // 0x3E
+    u32 OTID; // 0x48
+};
+
+struct UnkSummaryStruct
+{
+    /*0x00*/ union unkUnion *unk0;
+    /*0x04*/ void *unk4;
+    /*0x08*/ void *unk8;
+    /*0x0C*/ struct Pokemon currentPoke;
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Whenever you refer to Pokemon in shorthand, use Mon rather than Poke.

File: src/pokemon_summary_screen.c [hunk 9]
```diff
@@ -0,0 +1,3680 @@
...
+    u8 unk_filler5[0x19];
+    u8 unk40EF;
+    s16 unk40F0;
+    u8 unk_filler4[6];
+};
+
+void sub_81BF8EC(u8 a, void *b, u8 c, u8 d, void *e)
+{
+    u8 byte;
+    gUnknown_0203CF1C = AllocZeroed(0x40F8);
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Could this size be replaced with sizeof(UnkSummaryStruct)?

Comment (golemgalvanize) [reply src/pokemon_summary_screen.c]: nope, tried to

File: src/pokemon_summary_screen.c [hunk 10]
```diff
@@ -0,0 +1,3680 @@
...
+    u8 i;
+    for (i = 1; i < 4; i++)
+    {
+        if (gUnknown_0203CF1C->summary.moves[i] != 0)
+            return TRUE;
+    }
+    return FALSE;
+}
+
+void sub_81C4D18();
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: This forward declaration should go to the top.

File: src/pokemon_summary_screen.c [hunk 11]
```diff
@@ -0,0 +1,3679 @@
...
+                PlaySE(SE_SELECT);
+                gUnknown_0203CF21 = 4;
+                gSpecialVar_0x8005 = 4;
+                sub_81C044C(taskId);
+            }   
+        }
+    }
+}
+
+u8 sub_81C18A8()
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Seems like this should be a bool8 and the return value should be TRUE and FALSE.

File: src/pokemon_summary_screen.c [hunk 12]
```diff
@@ -0,0 +1,3679 @@
...
+            alloced[j+8] = 0x58;
+            alloced[j+9] = 0x59;
+        }
+    }
+    CopyToBgTilemapBufferRect_ChangePalette(3, alloced, 11, 0, 8, 2, 16);
+    schedule_bg_copy_tilemap_to_vram(3);
+    Free(alloced);
+}
+
+struct unkStruct_61CC04
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: This struct definition should be in PascalCase.

File: src/pokemon_summary_screen.c [hunk 13]
```diff
@@ -0,0 +1,3679 @@
...
+    }
+
+    r9 = &gUnknown_0203CF1C->unkTilemap1_1[0x255];
+    for (i = 0; i < 8; i++)
+    {
+        if (r6r7 > 7)
+            r9[i] = 0x206A;
+        else
+            r9[i] = 0x2062 + (r6r7 % 8);
+        r6r7
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: b should be a bool8.

Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: Are you sure r4 can't be a u16? It's being cast every time it's used.

Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: bool8

Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: bool8

Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: bool8

**Comment by Diegoisawesome** [comment]
Done reviewing, fix the comments and this PR can be accepted.

File: src/pokemon_summary_screen.c [hunk 14]
```diff
@@ -2998,7 +2998,7 @@ void sub_81C307C()
     sub_81AFC0C(0, gUnknown_0861CE74);
     sub_81AFC0C(1, gUnknown_0861CE7B);
     sub_81C31C0();
-    if (sub_81A6BF4() == 1 || sub_81B9E94() == 1 || sub_81C3304() == 1)
+    if (sub_81A6BF4() == 1 || sub_81B9E94() == 1 || sub_81C3304() == TRUE)
```
Comment (Diegoisawesome) [inline src/pokemon_summary_screen.c]: These two functions should also be bool8.

**Pull request merged**